src/charm.py

```diff
@@ -28,142 +42,101 @@ def __init__(self, msg: str, status_type=None):
-                    ]
-                },
-                "containers": [
-                    {
-                        "name": "kubeflow-dashboard",
-                        "imageDetails": image_details,
-                        "envConfig": {
-                            "USERID_HEADER": "kubeflow-userid",
-                            "USERID_PREFIX": "",
-                            "PROFILES_KFAM_SERVICE_HOST": f"{profiles_service}.{model}",
-                            "REGISTRATION_FLOW": config["registration-flow"],
-                            "DASHBOARD_LINKS_CONFIGMAP": config["dashboard-configmap"],
-                        },
-                        "ports": [{"name": "ui", "containerPort": config["port"]}],
-                        "kubernetes": {
-                            "livenessProbe": {
-                                "httpGet": {"path": "/healthz", "port": config["port"]},
-                                "initialDelaySeconds": 30,
-                                "periodSeconds": 30,
-                            }
-                        },
-                    }
-                ],
-            },
-            {
-                "configMaps": {
-                    config["dashboard-configmap"]: {
-                        "settings": json.dumps(
-                            {
-                                "DASHBOARD_FORCE_IFRAME": True,
-                            }
-                        ),
-                        "links": Path("src/config.json").read_text(),
-                    },
-                },
-                "kubernetesResources": {
-                    "customResources": {
-                        "profiles.kubeflow.org": [
-                            {
-                                "apiVersion": "kubeflow.org/v1beta1",
-                                "kind": "Profile",
-                                "metadata": {"name": config["profile"]},
-                                "spec": {
-                                    "owner": {"kind": "User", "name": config["profile"]}
-                                },
-                            }
-                        ]
+            self.framework.observe(self.on.remove, self._on_remove)
+
+    @property
+    def profiles_service(self):
+        return self._profiles_service
+
+    @profiles_service.setter
+    def profiles_service(self, service):
+        self._profiles_service = service
+
+    @property
+    def container(self):
+        return self._container
+
+    @property
+    def k8s_resource_handler(self):
+        if not self._k8s_resource_handler:
+            self._k8s_resource_handler = KubernetesResourceHandler(
+                field_manager=self._lightkube_field_manager,
+                template_files=self._resource_files.values(),
+                context=self._context,
+                logger=self.logger,
+            )
+        load_in_cluster_generic_resources(self._k8s_resource_handler.lightkube_client)
```

Comment:
Are we using kubeflow-dashboard specific CRs that we need to load? I can't seem to find them, but probably I am missing something.

Reply:
Nvm, I didn't realise `Profiles` was part of `kubeflow.org`